### PR TITLE
Update hugepages documentation

### PR DESCRIPTION
**Pull request by bart0sh**
- Described support of multiple huge page sizes
- Described container isolation
- Added description of the HugePageStorageMediumSize feature gate

### CODE REVIEW

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 1]
```diff
@@ -17,11 +17,11 @@ can consume huge pages and the current limitations.
 {{% capture prerequisites %}}
 
 1. Kubernetes nodes must pre-allocate huge pages in order for the node to report
-   its huge page capacity. A node may only pre-allocate huge pages for a single
-   size.
+   its huge page capacity. A node can pre-allocate huge pages for a multiple
+   sizes.
```
Comment (tengqm) [inline content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: ~a~ multiple sizes.

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 2]
```diff
@@ -17,11 +17,11 @@ can consume huge pages and the current limitations.
...
 
 The nodes will automatically discover and report all huge page resources as a
-schedulable resource.
+schedulable resources.
```
Comment (tengqm) [inline content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: ~a~

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 3]
```diff
@@ -30,12 +30,51 @@ schedulable resource.
 ## API
 
 Huge pages can be consumed via container level resource requirements using the
-resource name `hugepages-<size>`, where size is the most compact binary notation
-using integer values supported on a particular node. For example, if a node
-supports 2048KiB page sizes, it will expose a schedulable resource
-`hugepages-2Mi`. Unlike CPU or memory, huge pages do not support overcommit. Note
-that when requesting hugepage resources, either memory or CPU resources must
-be requested as well.
+resource name `hugepages-<size>`, where size is the most compact binary
```
Comment (tengqm) [inline content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]:
```suggestion
resource name `hugepages-<size>`, where `<size>` is the most compact binary
```

**Comment by tengqm** [comment]
This should target dev-1.18 branch.

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 4]
```diff
@@ -17,11 +17,11 @@ can consume huge pages and the current limitations.
 {{% capture prerequisites %}}
 
 1. Kubernetes nodes must pre-allocate huge pages in order for the node to report
-   its huge page capacity. A node may only pre-allocate huge pages for a single
-   size.
+   its huge page capacity. A node can pre-allocate huge pages for a multiple
```
[the remark this answers is not in the document]
Comment (liggitt) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: what is HugePageStorageMediumSize? I would not merge this documentation update until 1.19

[the remark this answers is not in the document]
Comment (bart0sh) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: @liggitt thank you for review. HugePageStorageMediumSize is a feature gate. It's disabled by default currently. @derekmahar suggested to implement it.

[the remark this answers is not in the document]
Comment (liggitt) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: is it in master? I don't see it

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md (hunk 1 above)
Comment (bart0sh) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: @tengqm thank you for the review.

**Comment by bart0sh** [comment]
@tengqm 
> This should target dev-1.18 branch.

Done

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bart0sh) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: It's in this PR: PR https://github.com/kubernetes/kubernetes/pull/84051
It's not merged yet as it's missed enhancement deadline. Other than that it's ready for approval. @derekwaynecarr reviewed it two times at least and doesn't have any objections.

[the remark this answers is not in the document]
Comment (bart0sh) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: @liggitt I should probably point out to this attempt to get an exception from the Release Team for hugepage enhancements. This is a tracking issue: https://github.com/kubernetes/enhancements/issues/1539. This PR is part of that activity.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 5]
```diff
@@ -103,6 +103,7 @@ different Kubernetes components.
 | `ExperimentalHostUserNamespaceDefaulting` | `false` | Beta | 1.5 | |
 | `EvenPodsSpread` | `false` | Alpha | 1.16 | |
 | `HPAScaleToZero` | `false` | Alpha | 1.16 | |
+| `HugePageStorageMediumSize` | `false` | Alpha | 1.18 | 1.19 |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]:
```suggestion
| `HugePageStorageMediumSize` | `false` | Alpha | 1.18 |  |
```

Comment (bart0sh) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: done

File: content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bg-chun) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: @liggitt 
We have requested an [release exception request](https://groups.google.com/forum/#!topic/kubernetes-sig-release/P2UdKZdau1Y) for 1.18 rel.

[And the exception is granted from sig-rel. ](https://github.com/kubernetes/enhancements/pull/1540#issuecomment-583072382)

For https://github.com/kubernetes/enhancements/issues/1539, it @derekwaynecarr approved days ago as the request is granted, but it seems approval from @thockin is also required.

[the remark this answers is not in the document]
Comment (bart0sh) [reply content/en/docs/tasks/manage-hugepages/scheduling-hugepages.md]: @liggitt [84051](https://github.com/kubernetes/kubernetes/pull/84051) has been merged. Would you mind reviewing/approving this PR now?

**Comment by liggitt** [comment]
This belongs to sig-node to review

**Comment by bart0sh** [comment]
@liggitt @derekwaynecarr All related PRs have been merged. Please, review/approve this one before the 1.18 deadline. Thanks.

**Comment by derekwaynecarr** [comment]
changes look good.

/approve
/lgtm

**Comment by bart0sh** [comment]
@kbhawkey can you remove hold, please? All code changes that this documentation update describes are already in the release-1.18 branch.

**Comment by bart0sh** [comment]
@sig-docs-en-owners this PR is ready for approval. Please, approve. Thanks.

**Comment by bart0sh** [comment]
@kubernetes/sig-docs-en-owners this PR is ready for approval. Please, approve. Thanks.

**Comment by sftim** [comment]
As this targets v1.18, it's the docs release team who would nomally approve.

**Comment by VineethReddy02** [comment]
@sftim Thanks for pulling me in. I had a quick look at this LGTM. :)

**Pull request merged**